test/crd/gateway.networking.k8s.io_gatewayclasses.yaml

```diff
@@ -0,0 +1,433 @@
+apiVersion: apiextensions.k8s.io/v1
```

Comment:
+1, let's please use the CRDs via upstream instead of keeping copies here.

Reply:
Hi @mlavacca and @shaneutt  sorry for the commented code will add the clean up part. One question so currently we are not maintaining a CRD folder in the config directory that's why I have created a separate folder for testing purpose. Can you please let me know from where I should pull the CRDs will update the code accordingly thanks.

Reply:
Yes the CRDs should be pulled from https://github.com/kubernetes-sigs/gateway-api, potentially using the `kustomize` directories there.

Reply:
@shaneutt I think we already have the integration test infra setup done.

Reply:
Do we need this PR now ?

Reply:
You know, now that you mention it I think I owe you an apology: I did not realize I ended up stepping all over your feet on this one. For some reason I got it in my head that this one would end up using the envtest stuff in the same directories as the controllers, similar to what kubebuilder will scaffold out for you.

I think we have something of an open question: envtest provides something a _little_ closer to unit tests (though obviously not true unit tests) than what we have in `test/integration` today. I personally can see the potential for value in having both, with the envtests hovering around the individual controller packages, and the integration tests being a more complete test of a real kubernetes environment (e.g. actually being able to communicate with real pods, e.t.c.), what do you think? Do you think we should push to have both? Alternatively we can use fake clients for a similar purpose :thinking:

Reply:
I think if we use fake clients that will be better. If our current integration testing code solving our needs then we can go with that only instead of adding envtest setup. Also if we include both of them we need to maintain them as well. So I think we can close this PR.

Reply:
Ok, sorry for the confusion and thank you for your help!

Reply:
No issues :)